components/log-viewer-webui/server/src/plugins/MongoReplicaServer/MongoReplicaServerCollection.ts

```diff
@@ -0,0 +1,141 @@
+import {
+    ChangeStream,
+    Collection,
+    Db,
+} from "mongodb";
+
+
+/**
+ * // eslint-disable-next-line no-warning-comments
+ * TODO: Improve this? Think about security (other queries should not be able to kick others
+ *  offline; maybe add a ref count then), performance, and collision chances.
+ *
+ * Generates a unique hash for a given query and options.
+ * This hash is used to identify and manage change streams for specific queries.
+ *
+ * @param query The query object to be hashed.
+ * @param options The options object to be hashed.
+ * @return A string representing the unique hash for the query and options.
+ */
+const getQueryHash = (query: object, options: object): string => JSON.stringify({query, options});
+
+interface Watcher {
+    // The change stream for the query
+    changeStream: ChangeStream;
+
+    // List of connection IDs that are subscribed to this watcher
+    subscribers: string[];
+}
+
+/**
+ * Class representing a MongoDB collection with support for change streams.
+ * This class manages subscriptions to real-time updates for specific queries.
+ */
+class MongoReplicaServerCollection {
+    // Reference count for active subscriptions
+    private count: number;
+
+    // MongoDB collection instance
+    private collection: Collection;
+
+    // Map of active change streams keyed by query hash
+    private watchers: Map<string, Watcher>;
```

Comment:
Again i think we can move this out of constructor
```suggestion
    private watchers: Map<string, Watcher> = new Map();
```